### PR TITLE
New Rule: Suspicious QR Code

### PR DESCRIPTION
**Pull request by morriscode**
I'm up for any suggestions on the name. This rule has been running in SharedEML's for a few weeks. Whenever we see a gap in our more brand specific QR code rules this rule has always gapped the coverage.

### CODE REVIEW

File: detection-rules/attachment_qr_code_with_highly_suspicious_content.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+source: |
+  type.inbound
+  and 0 < length(attachments) < 3
+  and all(attachments,
+          .file_type in $file_types_images
+          and (
+            any(file.explode(.),
+                .scan.qr.type == "url"
+                and (
+                  any([beta.linkanalysis(.scan.qr.url)],
```
[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_qr_code_with_highly_suspicious_content.yml:18]: I totally get where you're coming from, this rule has yet to FP but our sample set carries an obvious bias. The problem is the more specific we make this, the more we're chasing FN's over FP's.  The latter being less of a risk _usually_ ...

I'll see if I can come up with something that doesn't hurt overall effectiveness. Ultimately I'd like to see the gating happen within LA itself, but I understand the limitations and pitfalls there too. :/

File: detection-rules/attachment_qr_code_with_highly_suspicious_content.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                      or any(.files_downloaded,
+                             .file_extension in $file_extensions_common_archives
+                             or .file_extension in $file_extensions_executables
+                      )
+                  )
+                  or (
+                    .scan.qr.url.domain.root_domain in $url_shorteners
+                    and not strings.starts_with(.scan.qr.url.url, 'https://goo.gl/maps')
+                  )
+                  or .scan.qr.url.domain.root_domain == 'bing.com' and .scan.qr.url.path =~ '/ck/a'
```
Comment (jkamdjou) [inline detection-rules/attachment_qr_code_with_highly_suspicious_content.yml:32]: i think you're missing a parens around this `and`
```suggestion
                  or (
                      .scan.qr.url.domain.root_domain == 'bing.com' and .scan.qr.url.path =~ '/ck/a'
                  )
```

File: detection-rules/attachment_qr_code_with_highly_suspicious_content.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jkamdjou) [reply detection-rules/attachment_qr_code_with_highly_suspicious_content.yml:18]: do we think there are any one-time links in QR codes? like one-click confirmations, unsubscribes, etc? if so, we should gate some how. if not, may not be totally necessary

[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_qr_code_with_highly_suspicious_content.yml:18]: I tried OCR initially looking for 2FA, Scan with Phone, and other related words, but the same overlap is used in legitimate scenarios. Also observed some QR codes with no text, just the QR code itself. Going to toss up a bunch of gated or options that gets super close to covering the original set sans 2 outliers.

[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_qr_code_with_highly_suspicious_content.yml:35]: @ross lmkwyt

[the remark this answers is not in the document]
Comment (morriscode) [reply detection-rules/attachment_qr_code_with_highly_suspicious_content.yml:35]: Haha so sorry!

**Comment by morriscode** [comment]
Retiring i think this just got away from me, it's way over complicated. New replacement to be proposed

**Pull request closed without merging**